### ISSUE TITLE
Panic on program start

### ISSUE THREAD

**Issue by hreikin** [Julien-R44/fast-ssh#8]
When trying to start fast-ssh from the command line using `fast-ssh` it fails and freezes the terminal, I am trying this on Fedora 34 and have tested both the downloadable tar.gz file from the latest release (v0.2.1) as well as via the `cargo install fast-ssh` method. I have run the command with RUST_BACKTRACE=full as suggested and have provided the output below.

**EDIT:** Downgrading to an earlier version (v0.2.0) fixes the issue.

```
$ RUST_BACKTRACE=full fast-ssh > backtrace
thread 'main' panicked at 'index out of bounds: the len is 0 but the index is 0', /home/hreikin/.cargo/registry/src/github.com-1ecc6299db9ec823/fast-ssh-0.2.1/src/app.rs:67:10
                                                                                                                                                                               stack backtrace:
                                                                                                                                                                                                  0:     0x559d745f8b80 - <std::sys_common::backtrace::_print::DisplayBacktrace as core::fmt::Display>::fmt::hb81dd1a78bb96f60
                                                                                   1:     0x559d7461a5dc - core::fmt::write::h205940cafbb98a3d
                                                                                                                                                 2:     0x559d745f48f5 - std::io::Write::write_fmt::h4d30a65b89df279f
                                                                                                                                                                                                                        3:     0x559d745fa80b - std::panicking::default_hook::{{closure}}::h3c5425a3f4b5ee21
                                                                 4:     0x559d745fa380 - std::panicking::default_hook::h7c2827a5a7a96663
                                                                                                                                           5:     0x559d745faee4 - std::panicking::rust_panic_with_hook::hceee634f9d20b4f1
                                                                                                                                                                                                                             6:     0x559d745fa9b0 - std::panicking::begin_panic_handler::{{closure}}::h7de468b3df6750a8
                                                                             7:     0x559d745f9044 - std::sys_common::backtrace::__rust_end_short_backtrace::h87b00be28d10817a
                                                                                                                                                                                 8:     0x559d745fa919 - rust_begin_unwind
                                                                                                                                                                                                                             9:     0x559d74554291 - core::panicking::panic_fmt::h29a3e96afa0a0daa
                                                      10:     0x559d74554252 - core::panicking::panic_bounds_check::h8ecc421189d7d0f8
                                                                                                                                       11:     0x559d7456d623 - fast_ssh::app::App::get_items_based_on_mode::h189e8b590dbebaa5
                                                                                                                                                                                                                                12:     0x559d7455a0e5 - fast_ssh::widgets::hosts_widget::HostsWidget::render::ha6b60b69a568cd08
                                                                                    13:     0x559d7455f44a - tui::terminal::Terminal<B>::draw::hb733d0ac6f0e10f8
                                                                                                                                                                  14:     0x559d7456845e - <core::future::from_generator::GenFuture<T> as core::future::future::Future>::poll::ha27021561155e11e
                                                    15:     0x559d7455ca5f - tokio::park::thread::CachedParkThread::block_on::hf6c8bc90ccaeb030
                                                                                                                                                 16:     0x559d7455cbbe - tokio::runtime::thread_pool::ThreadPool::block_on::h7f84fdd4a7709496
                                                                                                                                                                                                                                               17:     0x559d7456ed38 - tokio::runtime::Runtime::block_on::h68ec888ceb8f60e5
                                                                                18:     0x559d7456dba4 - fast_ssh::main::he09aef1402a62e66
                                                                                                                                            19:     0x559d7455b6c3 - std::sys_common::backtrace::__rust_begin_short_backtrace::h74eefc0530100569
    20:     0x559d74569dad - std::rt::lang_start::{{closure}}::hedf0d1c98cec7b9e
                                                                                  21:     0x559d745fb4ea - std::rt::lang_start_internal::ha108655af4bf2480
                                                                                                                                                            22:     0x559d7456dc92 - main
                                                                                                                                                                                           23:     0x7ff65671eb75 - __libc_start_main
                                                                                                                                                                                                                                       24:     0x559d74554a0e - _start
                          25:                0x0 - <unknown>
```

**Comment by Julien-R44** [comment]
Hey thanks for reporting. Would you mind sending me your SSH config  with changed hostname ?

**Comment by Julien-R44** [comment]
I was able to reproduce the error with the following SSH config : 

```
Host *
  User hello

Host *.test.com
  User hello
```

But in this case Fast-SSH should just close with a message like "No host defined in your SSH config". Or maybe there is a use case I don't know about.
I guess you do have some hosts defined so I'll be curious to see your config, it would help me a lot !

**Comment by hreikin** [comment]
Hi mate, i love the tool i think its great. When i tried v0.2.1 initially my SSH config only contained the following:

```
Host *
    UpdateHostKeys yes
```

However that config above worked for v0.2.0 without any problems, since then i have defined some hosts and my config now looks like this:

```
Host *
    UpdateHostKeys yes
    
Host pihole
    HostName pi.hole
    User dietpi
    
Host owncloud
    HostName dietpi-owncloud.lan
    User dietpi
    
Host alarm
    HostName alarm.lan
    User alarm
    
Host debian
    HostName debian.lan
    User hreikin
```

I have just confirmed that this config works fine with both versions.

Another thing to note is i don't have a `config.yaml` file for fast-ssh anywhere on my system although this could maybe be a separate issue ? When i use `echo $XDG_CONFIG_HOME` it returns no output and if i run `sudo find / -name "config.yaml"` to search my system for the file it also finds nothing. Do you want me to open a separate issue for this ?

**Comment by Julien-R44** [comment]
Thanks man, I'm glad to hear that!
Ok cool, I'll add a clean error when the SSH configuration contains no valid entries

The configuration file is only available from version 0.3.0 which I released an hour ago. If you get it and run it you should be able to find your configuration file in ~/.config/FastSSH/config.yaml since you are on Fedora

**Comment by hreikin** [comment]
Awesome, thanks for the quick reply, i have just tried v0.3.0 and everything works perfectly and the `config.yaml` file is present on my system, thanks for the help!
